src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
     ObjectGuid _playerGUID;
 };
 
+enum SaltyRocka
+{
+    NPC_SALTY_GORGAR                            = 45497,
+
+    EVENT_ROCKA_CHECK_CONVERSATION              = 1,
+    EVENT_ROCKA_CHOOSE_CONVERSATION             = 2,
+    EVENT_ROCKA_CONVERSATION_COOLDOWN           = 3,
+    EVENT_ROCKA_TALK                            = 4,
+
+    TALK_ROCKA_0                                = 0,
+    TALK_ROCKA_1                                = 1,
+    TALK_ROCKA_2                                = 2,
+    TALK_ROCKA_3                                = 3,
+    TALK_ROCKA_4                                = 4,
+    TALK_ROCKA_5                                = 5,
+    TALK_ROCKA_6                                = 6,
+    TALK_ROCKA_7                                = 7,
+    TALK_ROCKA_8                                = 8,
+    TALK_ROCKA_9                                = 9,
+    TALK_GORGAR_0                               = 0,
+    TALK_GORGAR_1                               = 1,
+    TALK_GORGAR_2                               = 2,
+    TALK_GORGAR_3                               = 3,
+    TALK_GORGAR_4                               = 4,
+    TALK_GORGAR_5                               = 5
+};
+
+// 45498 - "Salty" Rocka
+struct npc_silverpine_salty_rocka : public ScriptedAI
+{
+    npc_silverpine_salty_rocka(Creature* creature) : ScriptedAI(creature), _playerNear(false) { }
+
+    void JustAppeared() override
+    {
+        CheckForGorgor();
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+        _events.ScheduleEvent(EVENT_ROCKA_CHECK_CONVERSATION, 1s);
+    }
+
+    void MoveInLineOfSight(Unit* who) override
+    {
+        if (who->GetTypeId() == TYPEID_PLAYER)
+            _playerGUID = who->GetGUID();
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ROCKA_CHECK_CONVERSATION:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        if (player->GetDistance2d(me) < 20.0f && !_playerNear)
+                        {
+                            _events.ScheduleEvent(EVENT_ROCKA_CHOOSE_CONVERSATION, 1s);
+                            _events.ScheduleEvent(EVENT_ROCKA_CONVERSATION_COOLDOWN, 180s);
+
+                            _playerNear = true;
+                        }
+                        else
+                        {
+                            _playerNear = false;
```

Comment:
setting it in `Reset()` should be enough instead of every time this event triggers

Reply:
Feedback applied.